react-components/src/utilities/getInstancesFromClick.ts

```diff
@@ -159,7 +160,7 @@ async function getAssetMappingPromise(
       ancestors
     );
 
-  return nodeAssetResult.mappings
-    .filter(isClassicCadAssetMapping)
-    .map((mapping) => ({ id: mapping.assetId }));
+  return nodeAssetResult.mappings.map((mapping) =>
+    instanceIdToInstanceReference(getMappingInstanceId(mapping))
```

Comment:
```
  if (isClassicCadAssetMappingInstance(mappingInstance)) {
    return mappingInstance.assetId;
  }

  if (isDmCadAssetMappingInstance(mappingInstance)) {
    return mappingInstance.instanceId;
  }
```

Do you expect you will ever come in MappingInstance? 

Because from what I see you map of those types
```
[{id: 123 } or { externalId: 'externalId', space: 'space' }]
```

Could it be that you also need ```assertNever```
Or did I miss something?

Reply:
@haakonflatval-cognite 
could it be that it never gets in the isClassicCadAssetMappingInstance or isDmCadAssetMappingInstance?